packages/openactive-integration-tests/test/helpers/flow-stages/flow-stage-recipes.js

```diff
@@ -118,6 +207,104 @@ const FlowStageRecipes = {
       defaultFlowStageParams,
     };
   },
+  /**
+   * A Recipe to either run B or P -> OrderFeedUpdate[SellerAcceptOrderProposalSimulateAction] -> B depending on the flow.
+   *
+   * - If in Simple Booking Flow, the recipe will just return B
+   * - If in Approval Flow, the recipe will return P -> OrderFeedUpdate[SellerAcceptOrderProposalSimulateAction] -> B
+   *
+   * This therefore represents the Book step of either flow.
+   *
+   * It is HIGHLY recommended that you use this function rather than manually creating B or P Flow Stages unless you
+   * are explicitly writing a test that uses one flow instead of another.
+   *
+   * The selected flow will be found in the `orderItemCriteriaList`, which will in almost all cases be based on the
+   * Booking Flow that is currently being tested. See (TODO) for more info about how Test Suite tests both Booking
+   * Flows.
+   *
+   * @param {OpportunityCriteria[]} orderItemCriteriaList
+   * @param {ReturnType<typeof FlowStageUtils.createDefaultFlowStageParams>} defaultFlowStageParams
+   * @param {object} args
+   * @param {UnknownFlowStageType} args.prerequisite
+   * @param {string | null} [args.brokerRole]
+   * @param {AccessPassItem[] | null} [args.accessPass]
+   * @param {PReqTemplateRef | null} [args.firstStageReqTemplateRef] Reference for the template which will be used
+   *   for the first stage - B or P.
+   * @param {() => import('./p').Input} args.getFirstStageInput Input for the first flow stage - B or P.
```

Comment:
P and B pretty much take the exact same types of requests. It says so here: https://openactive.io/open-booking-api/EditorsDraft/1.0CR3/#dfn-p:~:text=The%20details%20submitted%20are%20identical%20to%20those%20required%20for%20B%20with%20the%20addition%20of%20an%20optional%20lease.

Reply:
Also note that the template ref is a `PReqTemplateRef`. That's because the only difference between `PReqTemplateRef` and `BReqTemplateRef` is that the latter includes the `afterP` template which is exclusively used for the B at the end of approval flow. Therefore, regardless of flow, the first stage will never use `afterP`